cmake/linker/mwdt/target.cmake

```diff
@@ -0,0 +1,203 @@
+    set(linker_script_dep "")
+  endif()
+
+  zephyr_get_include_directories_for_lang(C current_includes)
+  get_filename_component(base_name ${CMAKE_CURRENT_BINARY_DIR} NAME)
+  get_property(current_defines GLOBAL PROPERTY PROPERTY_LINKER_SCRIPT_DEFINES)
+
+  add_custom_command(
+    OUTPUT ${linker_script_gen}
+    DEPENDS
+    ${LINKER_SCRIPT}
+    ${extra_dependencies}
+    # NB: 'linker_script_dep' will use a keyword that ends 'DEPENDS'
+    ${linker_script_dep}
+    COMMAND gcc
+    -x assembler-with-cpp
+    ${NOSYSDEF_CFLAG}
+    -MD -MF ${linker_script_gen}.dep -MT ${base_name}/${linker_script_gen}
+    -D_LINKER
+    -D_ASMLANGUAGE
+    ${current_includes}
+    ${current_defines}
+    ${linker_pass_define}
+    -E ${LINKER_SCRIPT}
+    -P # Prevent generation of debug `#line' directives.
+    -o ${linker_script_gen}
+    VERBATIM
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+  )
+endmacro()
+
+
+# Force symbols to be entered in the output file as undefined symbols
+function(toolchain_ld_force_undefined_symbols)
+  foreach(symbol ${ARGN})
+    zephyr_link_libraries(${LINKERFLAGPREFIX}-u${symbol})
+  endforeach()
+endfunction()
+
+# Link a target to given libraries with toolchain-specific argument order
+#
+# Usage:
+#   toolchain_ld_link_elf(
+#     TARGET_ELF             <target_elf>
+#     OUTPUT_MAP             <output_map_file_of_target>
+#     LIBRARIES_PRE_SCRIPT   [libraries_pre_script]
+#     LINKER_SCRIPT          <linker_script>
+#     LIBRARIES_POST_SCRIPT  [libraries_post_script]
+#     DEPENDENCIES           [dependencies]
+#   )
+function(toolchain_ld_link_elf)
+  cmake_parse_arguments(
+    TOOLCHAIN_LD_LINK_ELF                                     # prefix of output variables
+    ""                                                        # list of names of the boolean arguments
+    "TARGET_ELF;OUTPUT_MAP;LINKER_SCRIPT"                     # list of names of scalar arguments
+    "LIBRARIES_PRE_SCRIPT;LIBRARIES_POST_SCRIPT;DEPENDENCIES" # list of names of list arguments
+    ${ARGN}                                                   # input args to parse
+  )
+
+  target_link_libraries(
+    ${TOOLCHAIN_LD_LINK_ELF_TARGET_ELF}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_PRE_SCRIPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LINKER_SCRIPT}
+    ${TOOLCHAIN_LD_LINK_ELF_LIBRARIES_POST_SCRIPT}
+    ${LINKERFLAGPREFIX}-m
+    ${LINKERFLAGPREFIX}-Coutput=${TOOLCHAIN_LD_LINK_ELF_OUTPUT_MAP}
+    ${LINKERFLAGPREFIX}-Ball_archive
+    ${ZEPHYR_LIBS_PROPERTY}
+    ${LINKERFLAGPREFIX}-Bno_all_archive
+    kernel
+    $<TARGET_OBJECTS:${OFFSETS_LIB}>
+    ${LIB_INCLUDE_DIR}
+    -L${PROJECT_BINARY_DIR}
+    ${TOOLCHAIN_LIBS}
+
+    ${TOOLCHAIN_LD_LINK_ELF_DEPENDENCIES}
+  )
+endfunction(toolchain_ld_link_elf)
+
+macro(toolchain_ld_baremetal)
+# LINKERFLAGPREFIX comes from linker/ld/target.cmake
+  zephyr_ld_options(
+    -Hnocopyr
+    -Hnosdata
+    -Hnocrt
+    -Xtimer0 # to depress the warning message
```

Comment:
to suppress the warning message, we wouldn't want to depress it.

Reply:
-Xtimer0 in fact is a cpu related option,  mwdt requires at least one -Xxxx option, or warn msg will come out , so i add -Xtimer 0 here which is a common option for alll arc processors.

Reply:
I'm sure it's correct to add -Xtimer 0 here, but I'm saying that the comment should be saying "suppress", not "depress".